services/worker-manager/src/providers/aws.js

```diff
@@ -0,0 +1,398 @@
+const {ApiError, Provider} = require('./provider');
+const aws = require('aws-sdk');
+const _ = require('lodash');
+const taskcluster = require('taskcluster-client');
+const verify = require('iid-verify');
+
+class AwsProvider extends Provider {
+  constructor({
+    providerId,
+    monitor,
+    rootUrl,
+    Worker,
+    WorkerPool,
+    WorkerPoolError,
+    instancePermissions,
+    apiVersion,
+    region,
+    estimator,
+    validator,
+    notify,
+    ec2iid_DSA_SHA1_cert,
+  }) {
+    super({
+      providerId,
+      monitor,
+      rootUrl,
+      Worker,
+      WorkerPool,
+      WorkerPoolError,
+      estimator,
+      validator,
+      notify,
+    });
+    this.configSchema = 'config-aws';
+    this.instancePermissions = instancePermissions;
+    this.apiVersion = apiVersion;
+    this.region = region;
+    this.ec2iid_DSA_SHA1_cert = ec2iid_DSA_SHA1_cert;
+
+    aws.config.update({region});
```

Comment:
Different workerpools exist in different regions so I think we'll have to do this per-request ultimately?

Reply:
oh I forgot all about this, with all the iid stuff last week, thank you for reminding!